api/v1beta1/openstackvmset_types.go

```diff
@@ -42,8 +42,10 @@ type OpenStackVMSetSpec struct {
 	OSPNetwork Network `json:"ospNetwork"`
 	// Networks the name(s) of the OpenStackNetworks used to generate IPs
 	Networks []string `json:"networks"`
-	// Role the name of the Overcloud role this IPset is associated with. Used to generate hostnames.
+	// Role the name of the Overcloud role this RoleSpec is associated with. If it is a TripleO role, the name must match.
 	Role string `json:"role"`
+	// in case of external functionality, like 3rd party network controllers, set to false to ignore role in rendered overcloud templates.
+	IsTripleoRole bool `json:"isTripleoRole"`
```

Comment:
Could we use `kubebuilder` annotations to set a default of "true" and then have this be `omitempty`?  I'm just guessing that most of the time this will be set to true?

Reply:
yes that makes send, but to be able to use default values we need to switch to apiextensions.k8s.io/v1, default values can not be set using current apiextensions.k8s.io/v1beta1

~~~
Forbidden: must not be set (cannot set default values in apiextensions.k8s.io/v1beta1 CRDs, must use apiextensions.k8s.io/v1)
~~~

We can switch to that setting CRD_OPTIONS ?= "crd:crdVersions=v1" in the Makefile, but have to verify if it won't break anything else. In general I think this is be a good thing as we could set decent default values for our parameters.

https://github.com/operator-framework/operator-sdk/issues/3924
https://github.com/operator-framework/operator-sdk/issues/3580

Reply:
switch to v1 also sets preserveUnknownFields: false , which is good imo as it prevents accept CRs with typos in its name and fails early.

[1] https://kubernetes.io/blog/2019/06/20/crd-structural-schema/#pruning-don-t-preserve-unknown-fields

Reply:
done, I added a commit to move to apiextensions.k8s.io/v1, also set a default value for this boolena